packages/web/src/components/TextField/_TextField.scss

```diff
@@ -46,16 +49,50 @@
     width: 100%;
 }
 
-.TextField:hover > .TextField__input,
-.TextField__input:hover {
-    border-color: theme.$input-border-color-hover;
+.TextField__password-toggle {
+    position: relative;
+    display: flex;
 }
 
-.TextField:focus-within,
-.TextField:focus-within:hover,
-.TextField > .TextField__input:focus {
-    border-color: theme.$input-border-color-focus;
-    outline: 0;
+.TextField__password-toggle > .TextField__input {
+    flex-grow: 1;
+    padding-right: 0;
+    border-color: transparent;
+}
+
+.TextField__password-toggle__button {
+    @include reset.button();
+
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    padding: theme.$input-password-toggle-padding;
+
+    &::before {
+        content: '';
+        position: absolute;
+        top: 0;
+        right: 0;
+        bottom: 0;
+        left: 0;
+        border: theme.$input-border-width theme.$input-border-style theme.$input-border-color-default;
+        border-radius: theme.$input-border-radius;
+        pointer-events: none;
+    }
+}
+
+.TextField__password-toggle__icon {
```

Comment:
Maybe the time for Sass icon abstraction has come? It can be surely done in a separate PR.

Reply:
I will create an issue.